api/proto/teleport/legacy/types/types.proto

```diff
@@ -1661,6 +1663,57 @@ message ProvisionTokenSpecV2TerraformCloud {
   string Hostname = 3 [(gogoproto.jsontag) = "hostname,omitempty"];
 }
 
+message ProvisionTokenSpecV2Bitbucket {
+  // Rule is a set of properties the Bitbucket-issued token might have to be
+  // allowed to use this ProvisionToken.
+  message Rule {
+    // WorkspaceUUID is the UUID of the workspace for which this token was
+    // issued. Bitbucket UUIDs must begin and end with braces, e.g. '{...}'.
+    // This value may be found in the Pipelines -> OpenID Connect section of the
+    // repository settings.
+    string WorkspaceUUID = 1 [(gogoproto.jsontag) = "workspace_uuid,omitempty"];
+
+    // RepositoryUUID is the UUID of the repository for which this token was
+    // issued. Bitbucket UUIDs must begin and end with braces, e.g. '{...}'.
+    // This value may be found in the Pipelines -> OpenID Connect section of the
+    // repository settings.
+    string RepositoryUUID = 2 [(gogoproto.jsontag) = "repository_uuid,omitempty"];
+
+    // PipelineUUID is the UUID of the pipeline for which this token was issued.
+    // Bitbucket UUIDs must begin and end with braces, e.g. '{...}'
+    string PipelineUUID = 3 [(gogoproto.jsontag) = "pipeline_uuid,omitempty"];
+
+    // StepUUID is the UUID of the pipeline step for which this token was
+    // issued. Bitbucket UUIDs must begin and end with braces, e.g. '{...}'
+    string StepUUID = 4 [(gogoproto.jsontag) = "step_uuid,omitempty"];
+
+    // DeploymentEnvironmentUUID is the UUID of the deployment environment
+    // targeted by this pipelines run, if any. These values may be found in the
+    // "Pipelines -> OpenID Connect -> Deployment environments" section of the
+    // repository settings.
+    string DeploymentEnvironmentUUID = 5 [(gogoproto.jsontag) = "deployment_environment,omitempty"];
```

Comment:
Is this definitely the UUID? Their documentation gives some sample claims as :

```
{
  "sub": "{1de489be-ce6a-42a0-a8c8-eadbf1174ac7}:{bd715740-c970-486b-b68a-b421ec2a1f8b}:{759de0c6-eaee-4eaa-b7a6-c507eec759a7}",
  "aud": "ari:cloud:bitbucket::workspace/922e118c-f34f-4226-a897-0e0afbec553d",
  "stepUuid": "{759de0c6-eaee-4eaa-b7a6-c507eec759a7}",
  "deploymentEnvironment": "testing",
  "iss": "https://api.bitbucket.org/2.0/workspaces/bbcitest/pipelines-config/identity/oidc",
  "repositoryUuid": "{1de489be-ce6a-42a0-a8c8-eadbf1174ac7}",
  "branchName": "oidc-demo",
  "exp": 1578762300,
  "iat": 1578747600,
  "pipelineUuid": "{bee32f35-5dc0-4676-9ad1-657363be0ed0}",
  "workspaceUuid": "{922e118c-f34f-4226-a897-0e0afbec553d}"
}
```

If you've got a sample JWT to compare against, that'd be good.

Reply:
Yeah, their documentation isn't quite right. They give an example payload in the OIDC page in the repo settings which _also_ isn't quite right, but it at least does show deployment env UUIDs. This is the `.well-known/openid-configuration`:

```json
{
  "authorization_endpoint": "urn:bitbucket_cloud:programmatic_authorization",
  "claims_supported": [
    "sub",
    "iss",
    "aud",
    "exp",
    "iat",
    "branchName",
    "compliant",
    "deploymentEnvironmentUuid",
    "pipelineUuid",
    "repositoryUuid",
    "stepUuid",
    "workspaceUuid"
  ],
  "id_token_signing_alg_values_supported": [
    "RS256"
  ],
  "issuer": "https://api.bitbucket.org/2.0/workspaces/while-mountain-mercury/pipelines-config/identity/oidc",
  "jwks_uri": "https://api.bitbucket.org/2.0/workspaces/while-mountain-mercury/pipelines-config/identity/oidc/keys.json",
  "response_types_supported": [
    "id_token"
  ],
  "subject_types_supported": [
    "public"
  ]
}
``` 

And some parsed token claims extracted from an actual run:
```json
{
  "aud": "ari:cloud:bitbucket::workspace/2c6775bc-e495-4ea3-a2b0-afb2b0a47a6c",
  "branchName": "master",
  "deploymentEnvironmentUuid": "{c2d993ea-9d0b-48f9-9a1e-53d35f7329b1}",
  "exp": 1731458335,
  "iat": 1731449335,
  "iss": "https://api.bitbucket.org/2.0/workspaces/while-mountain-mercury/pipelines-config/identity/oidc",
  "pipelineUuid": "{0b016cd1-7d58-4b93-b5d6-5c03e0a47931}",
  "repositoryUuid": "{727730ef-d2cd-4978-ae6b-be160266ac29}",
  "stepUuid": "{ae834543-bf7f-4468-9fe6-071f20ee874c}",
  "sub": "{727730ef-d2cd-4978-ae6b-be160266ac29}:{c2d993ea-9d0b-48f9-9a1e-53d35f7329b1}:{ae834543-bf7f-4468-9fe6-071f20ee874c}",
  "workspaceUuid": "{2c6775bc-e495-4ea3-a2b0-afb2b0a47a6c}"
}
```

Reply:
Ok great - thanks TIm! A shame their docs are wrong...